infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/workflows/HealNodeWorkflow.java

```diff
@@ -27,11 +28,15 @@
 
     private final HealNodeRequest request;
 
-    public HealNodeWorkflow(HealNodeRequest healNodeRequest) {
-        super(new AddNodeRequest(healNodeRequest.getEndpoint()));
+    public HealNodeWorkflow(HealNodeRequest healNodeRequest, StreamLog streamLog) {
+        super(new AddNodeRequest(healNodeRequest.getEndpoint()), streamLog);
         this.request = healNodeRequest;
         this.actions = ImmutableList.of(new HealNodeToLayout(),
-                new RestoreRedundancyMergeSegments());
+                RestoreRedundancyMergeSegments.builder()
+                        .streamLog(streamLog)
+                        .currentNode(request.getEndpoint())
```

Comment:
For a sanity check, make sure that request.getEndpoint is equal to localEndpoint